### DISCUSSION TITLE
Oss-fuzz - possibility of increasing the amount of allocated RAM

### DISCUSSION THREAD

**Discussion by predator89090** [zeek/zeek#3895] [Help] [answered]
Hello, I would like to know if it is possible to increase the amount of allocated RAM for the zeek fuzzers for user`s plugins? By default, the -rss_limit_mb option is set to 2560, it is possible to increase it manually? There was a need to test for 2 hours, but at the end of one hour, an out of memory error occurred.

**Comment by awelzel** [comment, marked as answer]
Have you tried running the fuzzer with `-rss_limit_mb=xyz`  ?

```
$ ./build/src/fuzzers/zeek-ldap_tcp-fuzzer -help=1 2>&1 | head -30
Usage:
...
Flags: (strictly in form -flag=value)
 verbosity                              1       Verbosity level.
 seed                                   0       Random seed. If 0, seed is generated.
 runs                                   -1      Number of individual test runs (-1 for infinite runs).
[...]
 rss_limit_mb                           2048    If non-zero, the fuzzer will exit upon reaching this limit of RSS memory usage.
```

**Reply by predator89090** [reply]
I tried to run it with this option in oss-fuzz, but it didn't bring any results. 

![image](https://github.com/user-attachments/assets/9c2b37fd-37bc-4cb8-b597-632d31b4eae5)



Then I tried to build fuzzers and run them in zeek directly,  I followed the instructions in the readme, but I think it is outdated. Because after building a project with support for fuzzing and tests, some files are missing, like init-bare.zeek etc.

![image](https://github.com/user-attachments/assets/da870154-ac34-41a8-8b1a-d93b406b7a67)

After attempting to run fuzzer with corpus, I have an error "can't find base/init-bare.zeek"

**Reply by predator89090** [reply]
Still, I was able to run fuzzing in oss-fuzz with option -rss_limit_mb, but not through infra/helper.py run-fuzzer. I did this simply using the binary and the corpus directory. But I don't know how correct is this..
![image](https://github.com/user-attachments/assets/0c573523-f30c-48f1-9c5c-d7fba7dff848)
